### ISSUE TITLE
Hard-coded RPM repository filename element: incorrect (for Fedora 19/20) and brittle overall

### ISSUE THREAD

**Issue by aronr** [danieldreier/vagrant-template#6]
When I contributed a pull request to support installing Puppet and Facter on Fedora Linux recently, there was more than a malformed bash conditional expression in that request: there was also a problematic element - a hard-coded revision (?) number - in a URL pattern for retrieving Puppet Labs' RPM files that install package manager repos.

For two of the three most recent Fedora releases, as of this writing, that hard-coded number in the URL is incorrect, preventing that RPM file from being downloaded under Fedora 19 and 20.  (With wget's '--quiet' option set, this error is suppressed, so the install_puppet.sh script silently abends.)

At:
https://github.com/danieldreier/vagrant-template/blob/master/provision/install_puppet.sh#L138

The URL pattern given there is:
 REPO_URL="https://yum.puppetlabs.com/fedora/f${MAJOR_REV}/products/${MACH}/puppetlabs-release-${MAJOR_REV}-7.noarch.rpm"

However, the '7' in '-7.noarch.rpm' - which successfully identifies a downloadable file under Fedora 18 - should instead be either '2' or '10' for Fedora 19, and either '1' or '10' for Fedora 20.

I'm not sure what that number represents - it may indicate a revision number for the RPM files themselves - and if there may be a deterministic way to identify those numbers without resorting to hard-coding, such as within a conditional/case statement (and hence requiring ongoing maintenance of this script to reflect changes to the files in that repo).  Perhaps this might be an appropriate question for the relevant parties at Puppet Labs, who maintain those artifacts in the repo: whether they might consider using a 'latest' or 'current' alias or a similar file naming convention that could generate predictable URLs and help facilitate automation?

(This hard-coding is also present in the URL for RHEL; the number '7' in that URL happens to match actual filenames for both RHEL 5 and 6.)

**Comment by aronr** [comment]
Tangential but pertinent: challenges and approaches of getting wget to emit an error message only on failure:
http://superuser.com/questions/420120/wget-is-silent-but-it-displays-error-messages

**Comment by danieldreier** [comment]
Good catch. I'm working on it on the fedora1819support branch I created. I don't have access to a Fedora 19 VM so I can't readily test there, and just noticed that it seems to fail on CentOS 6.4 at the moment. I'll keep you updated.
